File: queuemate-ui/src/types/service.ts
export interface Service {
  id: string;
  businessId: string;
  name: string;
  description: string | null;
  durationMinutes: number;
  price: number;
  isActive: boolean;
  createdAtUtc: string;
  updatedAtUtc: string | null;
}

export interface CreateServiceRequest {
  name: string;
  description?: string;
  durationMinutes: number;
  price: number;
}

export interface UpdateServiceRequest {
  name: string;
  description?: string;
  durationMinutes: number;
  price: number;
  isActive: boolean;
}

File: QueueMate.Api/Model/Service.cs
using QueueMate.Api.Models.Common;

namespace QueueMate.Api.Models;

public sealed class Service : BaseEntity
{
    public Guid BusinessId { get; set; }

    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }

    public int DurationMinutes { get; set; }

    public decimal Price { get; set; }

    public bool IsActive { get; set; } = true;

    public Business Business { get; set; } = null!;
    public ICollection<StaffService> StaffServices { get; set; }
    = new List<StaffService>();
    public ICollection<Appointment> Appointments { get; set; }
    = new List<Appointment>();
}